src/mlpack/methods/ann/layer/atrous_convolution.hpp

```diff
@@ -0,0 +1,357 @@
+/**
+ * @file atrous_convolution.hpp
+ * @author Aarush Gupta
+ * @author Shikhar Jaiswal
+ *
+ * Definition of the Atrous Convolution class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_ATROUS_CONVOLUTION_HPP
+#define MLPACK_METHODS_ANN_LAYER_ATROUS_CONVOLUTION_HPP
+
+#include <mlpack/prereqs.hpp>
+
+#include <mlpack/methods/ann/convolution_rules/border_modes.hpp>
+#include <mlpack/methods/ann/convolution_rules/naive_convolution.hpp>
+#include <mlpack/methods/ann/convolution_rules/fft_convolution.hpp>
+#include <mlpack/methods/ann/convolution_rules/svd_convolution.hpp>
+
+#include "layer_types.hpp"
+
+namespace mlpack{
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the Atrous Convolution class. The Atrous Convolution
+ * class represents a single layer of a neural network.
```

Comment:
Would it be possible to add a little bit about what atrous convolution is and why someone would want it here?